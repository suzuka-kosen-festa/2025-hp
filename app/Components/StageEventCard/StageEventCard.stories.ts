import type { Meta, StoryObj } from "@storybook/react";

import StageEventCard from "./StageEventCard";

type T = typeof StageEventCard;

const meta = {
  title: "Components/StageEventCard",
  component: StageEventCard,
} satisfies Meta<T>;

export default meta;
type Story = StoryObj<T>;

export const Default: Story = {
  args: {
    title: "ステージイベント",
    datetime: "2025年11月2日(土) 10:00 - 11:00",
    stage: "メインステージ",
    description:
      "ステージイベントについての情報。",
  },
};
