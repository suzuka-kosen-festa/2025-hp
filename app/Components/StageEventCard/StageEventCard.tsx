import type { ReactNode } from "react";
import { AccessTime, Place } from "@mui/icons-material";
import {
  Box,
  Card,
  CardContent,
  CardMedia,
  Chip,
  Typography,
} from "@mui/material";

interface StageEventCardProps {
  image?: string
  title: string
  datetime: string
  stage: string
  description: string
  color: string
}

function StageEventCard({
  image,
  title,
  datetime,
  stage,
  description,
  color,
}: StageEventCardProps): ReactNode {
  const bgColor = `${color}33`;
  return (
    <Card
      sx={{
        "minWidth": 275,
        "transition": "0.3s",
        "&:hover": { boxShadow: 6 },
        "borderRadius": "16px",
        "border": `1px solid ${color}`,
        "color": color,
        "backgroundColor": bgColor,
        "height": "stretch",
      }}
    >
      {image && (
        <CardMedia
          component="img"
          image={image}
          alt={title}
          sx={{
            aspectRatio: "16/9",
            objectFit: "cover",
            borderTopLeftRadius: "16px",
            borderTopRightRadius: "16px",
          }}
        />
      )}
      <CardContent sx={{ "p": 2, "&:last-child": { pb: 2 } }}>
        <Typography variant="h5" component="div" gutterBottom sx={{ color: "white", fontWeight: "bold" }}>
          {title}
        </Typography>
        <Box
          sx={{
            display: "flex",
            alignItems: "center",
            color: "white",
            mb: 0.5,
          }}
        >
          <AccessTime sx={{ mr: 0.5, fontSize: "1rem", color: "white" }} />
          <Typography variant="body2" sx={{ color: "white" }}>{datetime}</Typography>
        </Box>
        <Box
          sx={{
            display: "flex",
            alignItems: "center",
            mb: 1.5,
          }}
        >
          <Place sx={{ mr: 0.5, fontSize: "1rem", color: "white" }} />
          <Chip label={stage} size="small" variant="outlined" sx={{ color: "white", borderColor: "white" }} />
        </Box>
        <Typography variant="body2" sx={{ color: "white" }}>
          {description}
        </Typography>
      </CardContent>
    </Card>
  );
}
export default StageEventCard;
