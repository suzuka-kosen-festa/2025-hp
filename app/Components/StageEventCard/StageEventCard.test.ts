// @vitest-environment jsdom

import { composeStories } from "@storybook/react";
import { expect, it } from "vitest";

import * as stories from "./StageEventCard.stories";

const { Default } = composeStories(stories);
it("should render a snapshot", async () => {
  await Default.run();
  expect(document.body.firstChild).toMatchSnapshot();
});
