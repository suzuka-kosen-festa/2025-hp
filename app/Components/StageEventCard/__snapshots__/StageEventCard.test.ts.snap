// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render a snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-uo3090-MuiPaper-root-MuiCard-root"
    style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiCardContent-root css-1vfyqek-MuiCardContent-root"
    >
      <div
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-124w7q-MuiTypography-root"
      >
        ステージイベント
      </div>
      <div
        class="MuiBox-root css-12liutp"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ay5ruu-MuiSvgIcon-root"
          data-testid="AccessTimeIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2M12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8"
          />
          <path
            d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body2 css-zsmq4j-MuiTypography-root"
        >
          2025年11月2日(土) 10:00 - 11:00
        </p>
      </div>
      <div
        class="MuiBox-root css-l69qyb"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ay5ruu-MuiSvgIcon-root"
          data-testid="PlaceIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7m0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5"
          />
        </svg>
        <div
          class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorDefault MuiChip-outlinedDefault css-1f8vrkm-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelSmall css-19m61dl-MuiChip-label"
          >
            メインステージ
          </span>
        </div>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body2 css-zsmq4j-MuiTypography-root"
      >
        ステージイベントについての情報。
      </p>
    </div>
  </div>
</div>
`;
